src/server/socketHandlers/mwPresenceSubscribe.js

```diff
@@ -0,0 +1,30 @@
+import parseChannel from './parseChannel';
+import {getTeamMember} from 'server/graphql/models/authorization';
+import {PRESENCE} from 'universal/subscriptions/constants';
+
+// const mockGetTeamMember = () => new Promise(resolve => {
+//   setTimeout(() => {
+//     resolve('hi');
+//   }, 2000);
+// });
+
+export default async function mwPresenceSubscribe(req, next) {
+  if (req.authTokenExpiredError) {
+    next(req.authTokenExpiredError);
+    return;
+  }
+  const {channel, variableString: meetingId} = parseChannel(req.channel);
+  if (channel !== PRESENCE) {
+    // all auth is taken care of inside GraphQL
+    next();
+    return;
+  }
+  const authToken = req.socket.getAuthToken();
+  // TODO cache all memberships on the socket?
+  const teamMember = await getTeamMember(authToken, meetingId);
```

Comment:
This makes things interesting. an async subscribe middleware means that we _must_ wait until we're sure we have a legit subscription before publishing something that we might want to listen for.